### PR TITLE
Switched to new sbt pr validator plugin

### PR DESCRIPTION
**Pull request by jroper**
This switches to the new sbt PR validator plugin.

The plugin has been substantially rewritten because it couldn't make the same assumptions about the code base structure and what needed to be run that Akka was making. I've tested it locally - when changing an individual project, it just builds that project and its dependencies, and when changing the project folder, it rebuilds all projects. Also, the plugin itself has reasonable scripted tests. But, it's difficult to comprehensively test it, so after merging, you should keep an eye on it to make sure it's doing the right thing.

Here are things that have changed to be aware of:

* Determining what changes are individual project changes, and what changes are sbt build changes, are no longer done based on simple matching of base directories, but rather using include and exclude filters. One consequence, this means changing `*.sbt` files in the root directory now triggers a full rebuild, since the filter for building all projects now matches these.
* Associating changes with projects is now done by inspecting the configured base directory of each project, and choosing the most specific one, rather than assuming Akka's flat project structure.
* Dependency analysis no longer uses the dependency graph plugin, but rather uses sbt's build structure directly. Not sure why the dependency graph plugin was used, since that requires going through Ivy, it's a lot simpler to just consider sbt's dependsOn configuration, which is what is done now.
* GitHub credentials now use sbt's credentials mechanism rather than an environment variable. To retrofit this into the existing build setup, I've configured the sbt credentials based on the environment variable.
* The new plugin supports both the Jenkins GHPRB plugin and Travis. Hopefully I've got the logic right, it's impossible to test it in a scripted test since you need the actual environment to make sure the integration is right.
* The new plugin does not port the `PrValidation` configuration, since that's specific to how Akka wants to do its PR validation (it wants to pass different options to the test framework). So that stuff still stays in Akka.
* The tasks that get run are now configurable via `prValidatorTasks`, `prValidatorBuildAllTasks` and `prValidatorEnforcedBuildAllTasks`.

Full documentation can be found here:

https://github.com/sbt/sbt-pull-request-validator

And now to make sure the GitHub enforced build integration is still working: PLS BUILD ALL

### CODE REVIEW

**Comment by jroper** [comment]
I should point out, I am fully expecting there to be bugs in this. One thing when I ran it locally, which I changed something in `akka-cluster-sharding-typed`, I didn't see paradox run (wasn't sure if it was supposed to), and I also didn't see MiMa run (also wasn't sure if it was supposed to). I did see the multi jvm tests run though.

File: project/ValidatePullRequest.scala
Comment (ktoso) [inline project/ValidatePullRequest.scala]: missing `ThisBuild / prValidatorBuildAllKeyword := "PLS BUILD ALL".r`

**Comment by ktoso** [comment]
Failure unrelated 


```
error] java.io.IOException: Error accessing /home/travis/.ivy2/cache/com.typesafe.akka/akka-stream_2.12/jars/akka-stream_2.12-2.5.6.jar
[error] 	at scala.reflect.io.FileZipArchive.scala$reflect$io$FileZipArchive$$openZipFile(ZipArchive.scala:140)
[error] 	at scala.reflect.io.FileZipArchive.x$2$lzycompute(ZipArchive.scala:177)
```

File: project/ValidatePullRequest.scala
Comment (ktoso) [reply project/ValidatePullRequest.scala]: Ah it is by default; the docs on the plugin made the impression it's the opposite. Should be good then

**Comment by ktoso** [comment]
PLS BUILD ALL

**Review by ktoso — APPROVED** [review]

**Comment by ktoso** [comment]
We'll also need to change akka-http and alpakka I think 

cc @2m @ennru

**Comment by jroper** [comment]
I don't have permission to click rebuild in travis, but apart from that, is there any reason why a 1.0 of this plugin shouldn't be released?

**Comment by raboof** [comment]
(restarted travis, the change sounds appealing but I haven't actually reviewed it yet)

**Comment by patriknw** [comment]
> changed something in akka-cluster-sharding-typed, I didn't see paradox run

We often have the snippet code in the project where they belong, because that is convenient when developing (not having to compile multiple projects). That probably means  that akka-docs doesn't have a dependency to akka-cluster-sharding-typed.  Changing the snippets could break paradox so perhaps we should always run akka-docs/paradox, or add that dependency, or is that too much overhead?

**Review by patriknw — COMMENTED** [review]
looking good, I guess.
Nice to simplify the Akka build and make this useful for others. After merging we should pay some attention to that pr validation is actually running the right things.

**Comment by ktoso** [comment]
> is there any reason why a 1.0 of this plugin shouldn't be released?

Sounds good; let's iterate on it in case we find anything to improve

**Comment by patriknw** [comment]
let's merge this after the release, to avoid risking more build issues

**Comment by raboof** [comment]
PLS BUILD

**Review by raboof — APPROVED** [review]
Nice to clean up, rebased because of conflict

**Comment by dwijnand** [comment]
LGTM.  Merge?

File: project/ValidatePullRequest.scala
Comment (dwijnand) [inline project/ValidatePullRequest.scala]: Should be set in global scope.

Comment (jroper) [reply project/ValidatePullRequest.scala]: Ok, I've made the change, but does it really matter? It's only ever read or overridden in the project scope.

**Comment by jroper** [comment]
Rebased and upgraded to version 1.0.0 of the pull request validator plugin. This should be good to be merged - though you might have to disable the branch protection for the typesafe-cla-validator, as it's currently failing due to an SSL issue with lightbend.com.

**Comment by jroper** [comment]
CLA validator is fixed.

File: project/ValidatePullRequest.scala
Comment (dwijnand) [reply project/ValidatePullRequest.scala]: Allows for it to be overridden in build scope.

It's also just a good habit to form, akin to not leaving a classes fields null after construction.

**Comment by dwijnand** [comment]
:tada:!

**Pull request merged**
